packages/react-async/README.md

```diff
@@ -135,54 +162,89 @@ const MyComponentOnIdle = createAsyncComponent({
 });
 ```
 
-#### Overwriting properties
+You can also pass a function to `defer`. This function, which will be called with the current props of the component, should return `true` when the component should begin loading. This makes it easy to implement components that have their visibility controlled by a property, like the Polaris Modal’s `open` prop:
+
+```tsx
+const MyModalComponent = createAsyncComponent({
+  load: () => import('./MyModalComponent'),
+  defer: ({open}) => open,
+});
+```
+
+#### Progressive hydration
 
-The library always allows you to pass an `async` prop with some custom options for the underlying `<Async />` loader component (**note**: this library reserves the `async` prop name, so you can’t use that name for any of your component’s own props, or for the props you specify in the `renderPreload`, `renderPrefetch`, or `renderKeepFresh` options).
+It can sometimes be useful to server render a component, but to wait to load its assets until later in the page lifecycle. This is particularly relevant for large, mostly static components, components that are very likely to be outside the viewport on load, and expensive components that would cause significant layout shifts if only rendered on the client. This library supports this pattern through the `deferHydration` option, and with the help from the [`@shopify/react-hydrate` package](../react-hydrate).
```

Comment:
Not sure how progressive hydration can solve layout shifts for this scenario:
- User loads a page with a component outside of the viewport at the bottom of the page. 
- Markup for this component is rendered on the server but CSS and JS assets are not loaded
- User scrolls to the bottom of the page. Assets are loaded.
- Component is re-rendered with its styles which might create a layout shift (CSS `height` is now set for example)

Reply:
Yeah, you're right, I need to make sure the styles load, but not the JS.